lib/grunt/deprecate.js

```diff
@@ -0,0 +1,116 @@
+var grunt = require('../grunt');
+
+function deprecate(obj, property, message) {
+  if (Array.isArray(obj)) {
+    obj.forEach(function(item) {
+      deprecate(item.obj, item.property, item.message);
+    });
+    return;
+  }
+  var logged = false;
+  function warn() {
+    var hideDeprecation = grunt.option('hide-deprecations');
```

Comment:
Can you look at `grunt.option('hide-deprecations')` at the top of `deprecate` and just `return` there if it's true?

Reply:
I put it in the warn to try to catch later calls to `grunt.option('hide-deprecations', true)`. So you could have the option to potentially hide warnings from other modules but not your own (prevent your team from using deprecated APIs but can't fix deprecations in other APIs).

But I might be complicating it unnecessarily, let me know and I'll move it out.